server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationFileCreator.java

```diff
@@ -206,14 +201,49 @@ public void copyUtilScripts() throws IOException {
         );
 
         logger.debug("Copying util scripts to the target artifact.");
-        for (String utilScript : utilScripts) {
-            if (!utilScript.isEmpty()) {
+        copyUtilFile(utilScripts, resourcePath, outputPath);
+    }
+
+    /**
+     Copies all needed cloudformation utility files to the target artifact.
+     Note: Theses are the files that actually need to be uploaded and accessed by EC2 instances unlike the util scripts.
+     */
+    public void copyUtilDependencies() throws IOException {
+        String resourcePath = "/cloudformation/files/util/";
+        String outputPath = "output/files/";
+
+        //Iterate over all files in the script list
+        logger.debug("Copying util files to the target artifact.");
+        copyUtilFile(cfnModule.getUtilFilesToBeUploaded(), resourcePath, outputPath);
+    }
+
+    /**
+     Adds file upload commands for all given files to the given script.
+     
+     @param files to be uploaded
+     @param script to add the file upload commands to
+     */
+    private void addFileUploadsToScript(List<String> files, BashScript script) {
+        files.forEach((filePath) -> {
+            String localFilePath = RELATIVE_DIRECTORY_PREFIX + filePath;
+            try {
+                script.append(uploadFile(filePath, localFilePath));
+            } catch (IOException e) {
+                throw new TransformationFailureException("Failed to add file uploads to the script.", e);
+            }
+        });
+    }
+    
+    public void copyUtilFile(List<String> files, String resourcePath, String outputPath) throws IOException {
+        PluginFileAccess fileAccess = cfnModule.getFileAccess();
+        for (String file : files) {
+            if (!file.isEmpty()) {
                 //Copy the file into the desired directory
-                logger.debug("Adding " + utilScript + " to the target artifact.");
+                logger.debug("Adding " + file + " to the target artifact.");
```

Comment:
Please consider using `logger.debug("Adding {} to the target artifact.", file);`

This will only concatenate if the loglevel requires the rendering of the message (i.e. level is set to debug or trace)